keps/sig-api-machinery/2876-crd-validation-expression-language/README.md

```diff
@@ -589,6 +590,29 @@ associativeList.all(e, e.key1 == 'a' && e.key2 == 'b' && e.val == 100)
 associativeList.all(e, e.val == 100)
 ```
 
+### Resource constraints
+
+For Beta, per-request execution time will be constrained via a context-passed timeout. For Beta, we are looking at a range between 100 milliseconds and 1 second for the timeout. We want to see how
+CEL performs in benchmarks once our Beta optimizations are in place before deciding what specific 
+duration to set the timeout to. If the timeout is  exceeded, the error message will include how
+long each expression took to evaluate.
```

Comment:
this has the downside of permitting costly CEL expressions that time out to be persisted in the CRD, and fail at CR request time

Reply:
Definitely - only having a timeout doesn't provide nearly as strong guarantees as estimating expression cost. I think if we can guarantee good cost estimation, we should go with it. I'm a little concerned about accuracy, since I feel there could be cases where an incorrect estimate is given, and an expression that should be allowed isn't permitted (or vice versa).